tianshou/policy/imitation/cql.py

```diff
@@ -167,6 +173,35 @@ def calc_random_values(
 
         return random_value1 - random_log_prob1, random_value2 - random_log_prob2
 
+    def process_buffer(self, buffer: ReplayBuffer | None) -> ReplayBuffer | None:
+        if buffer is None:
```

Comment:
When would that be the case? The whole logic of calibrated=True is not applied then. If there are such cases, one should either accommodate them, or raise an informative error if `self.calibrated=True`